firmware/boot/bootloader.c

```diff
@@ -97,37 +94,28 @@ _Noreturn static void modifyStackPointerAndStartApp(const uint32_t *address)
 
     // Should never get here!
     BREAK_IF_DEBUGGER_CONNECTED()
-    for (;;)
-    {
+    for (;;) {
     }
 }
 
-static BootStatus verifyAppCodeChecksum(void)
-{
-    if (*&__app_code_start__ == 0xFFFFFFFF)
-    {
+static BootStatus verifyAppCodeChecksum(void) {
+    if (__app_code_start__ == 0xFFFFFFFF) {
```

Comment:
I vaguely remember you can't do this with linker script variables, and you have to keep the `*&`. I might be wrong but you should try this on hardware.